Nishakm/listen for callback

nishakm opened the pull request.

nishakm committed:
Register with webhook and listen for callback

Signed-off-by: nisha <nishakumarx@gmail.com>

Commit:
Gracefully close the socket server

Signed-off-by: Nisha Kumar-Mayernik <nkumarm1@jarguarlandrover.com>

Commit:
Set socket timeout to 15 minutes

Signed-off-by: Nisha Kumar-Mayernik <nkumarm1@jarguarlandrover.com>

prossaro commented on src/main/java/org/jenkinsci/plugins/nouvoladivecloud/NouvolaBuilder.java:
```diff
@@ -45,63 +47,175 @@ public String getPlanID() {
+        String urlParameters = "creds_pass=" + Secret.toString(credsPass);
+
+        String registerUrl   = "http://preproduction.nouvola.com/api/v1/hooks";
+        String triggerUrl = "http://preproduction.nouvola.com/api/v1/plans/" + planID + "/run";
```
The path for the product is: https://divecloud.nouvola.com/api/v1/...

Commit:
Added changes to UI

- Users can now enter a timeout in minutes
Defaults to 60 minutes if not entered
- The listener can now listen on any port
Defaults to 9999 if no port exists in the url

Signed-off-by: Nisha Kumar-Mayernik <nkumarm1@jarguarlandrover.com>

nishakm commented:
Pull request needs to be accepted in order for it to be merged to master.

prossaro reviewed (approved):
This looks good!
